crates/common-domain/src/ids/mod.rs

```diff
@@ -0,0 +1,170 @@
+use crate::id_type;
+use sealed::sealed;
+use std::error::Error;
+use std::fmt::{Debug, Display};
+use std::ops::Deref;
+use std::str::FromStr;
+use uuid::Uuid;
+
+mod alias_or;
+mod macros;
+
+pub use alias_or::AliasOr;
+
+id_type!(CustomerId, "cus_");
+
+#[derive(Debug)]
+pub struct IdError(pub(crate) String);
+impl Display for IdError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "IdError: {}", self.0)
+    }
+}
+
+impl Error for IdError {}
+
+#[sealed]
+pub trait BaseId: Deref<Target = Uuid> {
+    const PREFIX: &'static str;
+    type IdType;
+
+    fn new() -> Self::IdType;
+    fn as_uuid(&self) -> Uuid {
+        **self
+    }
+    fn parse_uuid(s: &str) -> Result<Self::IdType, IdError>;
+}
+
+pub mod string_serde {
```

Comment:
this serde should be used in the apis, currently in rest_api as we use there json serde.